src/main/java/reactor/ipc/netty/channel/ChannelOperationsHandler.java

```diff
@@ -458,7 +462,17 @@ else if (future instanceof ChannelPromise) {
 						else {
 							innerActive = true;
 							inner.promise = promise;
-							p.subscribe(inner);
+							AttributeKey<Object> attributeKey = AttributeKey.valueOf("subscriberContext");
+
+							if(ctx.channel()
+									.hasAttr(attributeKey)) {
+								p.subscribe(new ContextStartSubscriber<>(inner, inner.currentContext()
```

Comment:
instead of a new ContextStartSubscriber why not Flux.from(p).subscribe(inner, currentContext.putAll(attr.get()))

Reply:
Hah, will fix it, I did not know about that method :)